### PR TITLE
Garage/reel

### PR DESCRIPTION
**Pull request by austin-hwang**
### Description
Initial pull request to merge Xamarin.Forms app into the recommenders repository.  Recommenders Engine Example Layout (REEL) is a movie recommendation cross-platform application that demonstrates the integration of some of recommendation algorithms on Microsoft/Recommenders into a cross-platform application workflow (Android, iOS, UWP, HoloLens). Additional details can be found in the README in the root directory of REEL.

### Checklist:
- [x] I have followed the [contribution guidelines](../CONTRIBUTING.md) and code style for this project.
- [x] I have added tests covering my contributions.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: notebooks/05_operationalize/aks_locust_load_test.ipynb [hunk 1]
```diff
@@ -114,34 +114,26 @@
    "outputs": [],
```
Comment (gramhagen) [inline notebooks/05_operationalize/aks_locust_load_test.ipynb]: was this change intentional?

File: examples/REEL/backend/data/movies_by_genre.json [hunk 2]
```diff
@@ -0,0 +1,254 @@
+{
```
[the remark this answers is not in the document]
Comment (austin-hwang) [reply examples/REEL/backend/data/movies_by_genre.json]: This file is used by the /onboarding route because we just hard-coded some popular movies that will be suggested since we don't have an sort of popularity metric in the database.

File: examples/REEL/backend/data/starter_movies.json [hunk 3]
```diff
@@ -0,0 +1,1319 @@
+{
```
[the remark this answers is not in the document]
Comment (austin-hwang) [reply examples/REEL/backend/data/starter_movies.json]: Same here, we use this file for the browse page where we provide popular movies for each genre. I'll double check about the tmdb info.

File: notebooks/05_operationalize/aks_locust_load_test.ipynb (hunk 1 above)
Comment (Brucegatete) [reply notebooks/05_operationalize/aks_locust_load_test.ipynb]: Which change is this comment commented on?

File: examples/REEL/backend/README.md [hunk 4]
```diff
@@ -0,0 +1,290 @@
...
+
+## Install
+
+```sh
+pip install -r requirements.txt
+```
+
+## Setting up ML model endpoint
+
+Run the desired jupyter notebook (either SAR or Light GBM) and write down the REST url endpoint and the Bearer token returned at the end of the notebook by the following cell:
```
[the remark this answers is not in the document]
Comment (austin-hwang) [reply examples/REEL/backend/README.md]: Fixed

File: examples/REEL/data/populateMovieData.py [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+import sys
+
+"""
+Populates the database with information about each movie such as there movieID, title, date, and genres
+Reads this information through a dataset downloaded from MovieLens
+Data Format:
+    1) COUNT,MOVIE_ID,MOVIE_TITLE MOVIE_YEAR,GENRE|GENRE|GENRE
+    2) COUNT,MOVIE_ID,"MOVIE_TITLE MOVIE_YEAR",GENRE|GENRE|GENRE - there is a comma and/or parenthesis
+
+@author: Blake Wesel
```
[the remark this answers is not in the document]
Comment (austin-hwang) [reply examples/REEL/data/populateMovieData.py]: Fixed

File: examples/REEL/data/tmdbIDPopulation.py [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+import pyodbc
+import os
+import time
+import sys
+
+"""
+Adds a column to the MOVIES table called tmdbID and populates it with every movies' correct tmdbID
+Command to add the new column: ALTER TABLE MOVIES ADD tmdbID int;
+
+@author: Blake Wesel
```
[the remark this answers is not in the document]
Comment (austin-hwang) [reply examples/REEL/data/tmdbIDPopulation.py]: Fixed

File: examples/REEL/mobile/README.md [hunk 7]
```diff
@@ -0,0 +1,25 @@
+# Introduction
+This mobile application was built using Xamarin.Forms which is an open source framework for building iOS, Android, and Windows apps. Learn more about Xamarin Forms platform at https://www.xamarin.com/forms.
+
+# Installation Guide
+Pre-requisite: the backend web service is either running locally on your computer, or is deployed to AKS
+
+1. Ensure your version of Visual Studio has workspace for developing Xamarin.Forms applications. If it does not, launch the Visual Studio Installer and install the Mobile development with .NET workload (https://docs.microsoft.com/en-us/xamarin/get-started/installation)
```
[the remark this answers is not in the document]
Comment (austin-hwang) [reply examples/REEL/mobile/README.md]: Fixed

File: examples/REEL/.gitignore [hunk 8]
```diff
@@ -0,0 +1,362 @@
+## Ignore Visual Studio temporary files, build results, and
```
[the remark this answers is not in the document]
Comment (austin-hwang) [reply examples/REEL/.gitignore]: Merged

File: examples/REEL/README.md [hunk 9]
```diff
@@ -0,0 +1,45 @@
+![alt text](backend/assets/reel_mock.png =500x)
+## Introduction 
+
+Recommenders Engine Example Layout (REEL) is a movie recommendation cross-platform mobile application that demonstrates the integration of some of recommendation algorithms on Microsoft/Recommenders into a mobile application workflow.
```
[the remark this answers is not in the document]
Comment (austin-hwang) [reply examples/REEL/README.md]: Fixed

File: examples/REEL/README.md [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+The following sections will demonstrate how to locally build REEL. These sections will guide the reader into:
+	
+* Creating an ML model endpoint
+* Setting up and populating [Azure SQL Database](https://docs.microsoft.com/en-us/azure/sql-database/)
+* Setting up [Azure Search](https://docs.microsoft.com/en-us/azure/search/)
+* Running the app's backend on Azure using [AKS](https://docs.microsoft.com/en-us/azure/aks/)
+* Installing the [Xamarin.Forms](https://docs.microsoft.com/en-us/xamarin/xamarin-forms/) app for the mobile client
+	
+Currently, REEL runs *Simple Algorithm for Recommendation (SAR)* and *LightGBM algorithms*. The application is built using [Xamarin.Forms](https://docs.microsoft.com/en-us/xamarin/xamarin-forms/),
+ so it is supported on **iOS**, **Android** and **UWP**. The algorithms are trained on the [MovieLens dataset](https://grouplens.org/datasets/movielens/).
```
[the remark this answers is not in the document]
Comment (austin-hwang) [reply examples/REEL/README.md]: Fixed

File: examples/REEL/data/populateMovieData.py [hunk 11]
```diff
@@ -0,0 +1,136 @@
...
+@version: 8/20/19
```
[the remark this answers is not in the document]
Comment (austin-hwang) [reply examples/REEL/data/populateMovieData.py]: Done

File: notebooks/05_operationalize/aks_locust_load_test.ipynb (hunk 1 above)
Comment (gramhagen) [reply notebooks/05_operationalize/aks_locust_load_test.ipynb]: this is regarding the locust load testing notebook

Comment (daranzhao) [reply notebooks/05_operationalize/aks_locust_load_test.ipynb]: I made some small changes with where the path to the config file could be found. The big change of deleting the 2 cells that verify the service was because they threw errors when I ran them with our two endpoints.

Comment (gramhagen) [reply notebooks/05_operationalize/aks_locust_load_test.ipynb]: i don't think we need to capture the application of this notebook for load testing the app, we should leave it as it was (or provide an update on how to set up the config file if that part led to confusion).

Comment (daranzhao) [reply notebooks/05_operationalize/aks_locust_load_test.ipynb]: Ah makes sense, I'll remove the notebook from here

**Comment by roalexan** [comment]
I needed to change as follows in sar_webservice_poc.ipynb
#os.chdir('~/recommenders/notebooks/00_quick_start')
os.chdir('C:\\Users\\rba1\\Recommenders\\notebooks\\00_quick_start')
we should make this platform independent

**Comment by roalexan** [comment]
Probably not a huge deal, but the last "cleanup" cell didn't work for me. The first problem was that it could find the temp user folder. Note that I kicked off creating the AKS cluster, then continued the next day. When I updated CURRENT_DIR, I then get a permission error stating that a file is in use by another file. So, don't think this should hold off the PR, but wanted to note this.

File: examples/REEL/README.md [hunk 12]
```diff
@@ -0,0 +1,101 @@
+<img src="assets/reel_mock.png">
```
Comment (miguelgfierro) [inline examples/REEL/README.md]: I don't think we should add another folder `examples`, that would increase the complexity of the folder structure. One of the ideas that we were considering is to rename script folder to tools. Do you think tools would makes sense for this content?

Comment (miguelgfierro) [reply examples/REEL/README.md]: yeah, maybe a separate repo would make sense. We can discuss it tomorrow

**Comment by gramhagen** [comment]
closed pending review of where best to land this
